AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_entity_utils.py

```diff
@@ -313,6 +313,42 @@ def get_type_ids(component_names: list, entity_type: EditorEntityType = EditorEn
             bus.Broadcast, "FindComponentTypeIdsByEntityType", component_names, entity_type.value)
         return type_ids
 
+    def is_property_visible(self, component_property_path: str, expected: bool = True) -> bool:
+        """
+        Used to determine if a property is visible.
+        param component_property_path: String of component property. (e.g. 'Settings|Visible')
+        :param expected: The expected visibility (default True)
+        :return: None
+        """
+        visible_key = 1
+
+        get_component_property_outcome = self.get_property_type_visibility()
+        result = get_component_property_outcome[component_property_path][visible_key]
+
+        assert (result == "Visible") == expected, \
+            f"Failure: Property visibility was not {expected} for" \
+            f"{self.get_component_name()} : {component_property_path}"
+
+        return result
+
+    def toggle_property_switch(self, component_property_path: str) -> None:
+        """
+        Used to toggle a property switch
+        param component_property_path: String of component property. (e.g. 'Settings|Visible')
+        :return: None
+        """
+        start_value = self.get_component_property_value(component_property_path)
+        if start_value:
+            self.set_component_property_value(component_property_path, False)
+        else:
+            self.set_component_property_value(component_property_path, True)
```

Comment:
This function, as written, intends to just flip the current state as a user would toggle it. However, I have considered an override that says you intend to toggle to true and validates if that's possible. Otherwise we'd just use the explicit set.

Reply:
Moving this toggle to validation behavior to another helper to make thing clear on intent and to decouple it from `EditorComponent` class.